hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestDirectoryDeletingServiceWithFSOBucket.java

```diff
@@ -315,4 +318,60 @@ private void checkPath(Path path) {
     }
   }
 
+  @Test
+  public void testDeleteSubFiles() throws Exception {
+
+    Table<String, OmKeyInfo> deletedDirTable =
+        cluster.getOzoneManager().getMetadataManager().getDeletedDirTable();
+    Table<String, OmKeyInfo> keyTable =
+        cluster.getOzoneManager().getMetadataManager().getKeyTable();
+    Table<String, OmDirectoryInfo> dirTable =
+        cluster.getOzoneManager().getMetadataManager().getDirectoryTable();
+    Table<String, RepeatedOmKeyInfo> deletedKeyTable =
+        cluster.getOzoneManager().getMetadataManager().getDeletedTable();
+
+    Path root = new Path("/rootDir2");
+    // Create  parent dir from root.
+    fs.mkdirs(root);
+
+    // Added 10 sub files inside root dir
+    for (int i = 0; i<10; i++) {
+      Path path = new Path(root, "testKey" + i);
+      try (FSDataOutputStream stream = fs.create(path)) {
+        stream.write(1);
+      }
+    }
+
+    // Before delete
+    assertTableRowCount(deletedDirTable, 0);
+    assertTableRowCount(keyTable, 10);
+    assertTableRowCount(dirTable, 1);
+
+    fs.delete(root, true);
+
+    DirectoryDeletingService dirDeletingService =
+        (DirectoryDeletingService) cluster.getOzoneManager().getKeyManager()
+            .getDirDeletingService();
+
+    KeyDeletingService keyDeletingService =
+        (KeyDeletingService) cluster.getOzoneManager().getKeyManager()
+            .getDeletingService();
+
+    // Just After delete
+    assertTableRowCount(deletedDirTable, 1);
+    assertTableRowCount(keyTable, 0);
+    assertTableRowCount(dirTable, 0);
+    assertTableRowCount(deletedKeyTable, 10);
+
+    // Eventually keys would get cleaned up from deletedTables too
+    assertTableRowCount(deletedDirTable, 0);
+    assertTableRowCount(deletedKeyTable, 0);
+
+    assertSubPathsCount(dirDeletingService.getMovedFilesCount(), 10);
+    assertSubPathsCount(dirDeletingService.getDeletedDirsCount(), 1);
+    // verify whether KeyDeletingService has purged the keys
+    Assert.assertEquals(10, keyDeletingService.getDeletedKeyCount().get());
+  }
+
+
```

Comment:
Please add following test case to verify the direct file deletion.


  @Test
  public void testDeleteFiles() throws Exception {

    Table<String, OmKeyInfo> deletedDirTable =
        cluster.getOzoneManager().getMetadataManager().getDeletedDirTable();
    Table<String, OmKeyInfo> keyTable =
        cluster.getOzoneManager().getMetadataManager().getKeyTable();
    Table<String, OmDirectoryInfo> dirTable =
        cluster.getOzoneManager().getMetadataManager().getDirectoryTable();
    Table<String, RepeatedOmKeyInfo> deletedKeyTable =
        cluster.getOzoneManager().getMetadataManager().getDeletedTable();

    Path root = new Path("/rootDir2");
    // Create  parent dir from root.
    fs.mkdirs(root);

    // Added 10 sub files inside root dir
    for (int i = 0; i<10; i++) {
      Path path = new Path(root, "testKey" + i);
      try (FSDataOutputStream stream = fs.create(path)) {
        stream.write(1);
      }
    }

    // Before delete
    assertTableRowCount(deletedDirTable, 0);
    assertTableRowCount(keyTable, 10);
    assertTableRowCount(dirTable, 1);

    for (int i = 0; i<10; i++) {
      Path path = new Path(root, "testKey" + i);
      fs.delete(path, true);
    }

    DirectoryDeletingService dirDeletingService =
        (DirectoryDeletingService) cluster.getOzoneManager().getKeyManager()
            .getDirDeletingService();

    KeyDeletingService keyDeletingService =
        (KeyDeletingService) cluster.getOzoneManager().getKeyManager()
            .getDeletingService();

    // Just After delete
    assertTableRowCount(keyTable, 0);
    assertTableRowCount(dirTable, 1);

    // Eventually keys would get cleaned up from deletedTables too
    assertTableRowCount(deletedDirTable, 0);
    assertTableRowCount(deletedKeyTable, 0);

    assertSubPathsCount(dirDeletingService.getMovedFilesCount(), 0);
    assertSubPathsCount(dirDeletingService.getDeletedDirsCount(), 0);
    // verify whether KeyDeletingService has purged the keys
    Assert.assertEquals(10, keyDeletingService.getDeletedKeyCount().get());
  }

Reply:
Inorder to run the direct file deletion, we need to do correct keyName in `OMKeyDeleteResponseV1` similar way you did in OMPathsPurgeResponseV1.java class

https://github.com/apache/ozone/blob/HDDS-2939/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyDeleteResponseV1.java#L81

 ```
     Table<String, OmKeyInfo> keyTable = omMetadataManager.getKeyTable();
      OmKeyInfo omKeyInfo = getOmKeyInfo();
      // Sets full absolute key name to OmKeyInfo, which is
      // required for moving the sub-files to KeyDeletionService.
      omKeyInfo.setKeyName(keyName);
      String deletedKey = omMetadataManager
          .getOzoneKey(omKeyInfo.getVolumeName(), omKeyInfo.getBucketName(),
              omKeyInfo.getKeyName());
      // For OmResponse with failure, this should do nothing. This method is
      // not called in failure scenario in OM code.
      keyTable.deleteWithBatch(batchOperation, ozoneDbKey);
      addDeletionToBatch(omMetadataManager, batchOperation, keyTable,
          deletedKey, omKeyInfo);
```

Reply:
Also, please add `DELETED_DIR_TABLE` to the OMKeyDeleteResponseV1 class.

```
/**
 * Response for DeleteKey request.
 */
@CleanupTableInfo(cleanupTables = { FILE_TABLE, DIRECTORY_TABLE, DELETED_TABLE,
    DELETED_DIR_TABLE })
public class OMKeyDeleteResponseV1 extends OMKeyDeleteResponse {
```

Reply:
Done.